common/djangoapps/student/views.py

```diff
@@ -1576,6 +1590,8 @@ def create_account_with_params(request, params):
                 user.email_user(subject, message, from_address)
         except Exception:  # pylint: disable=broad-except
             log.error(u'Unable to send activation email to user from "%s"', from_address, exc_info=True)
+    else:
+        registration.activate()
```

Comment:
I'm concerned that this is a new else-clause that will be executed in cases that pre-existed.  That is, even without setting SKIP_EMAIL_VALIDATION, this code changes the existing behavior of the system.  One is a testing-only condition, and the other is extauth, so I think they might be OK, but I'd like to understand more about what those settings are for.